### PR TITLE
Regional capabilities for precip variability

### PR DESCRIPTION
**Pull request by acordonez**
- [x] Subset grid by latitude and longitude - CropLatLon function and parser updates
- [x] Add "standard" calendar checks
- [x] Update filenames for arbitrary grid size
- [x] Update RegridHoriz function for arbitrary lat/lon resolution
- [x] Update Avg_PS_DomFrq to add custom region option
- [x] Add option to crop dataset via shapefile
- [x] Check for any remaining cdat reference to remove
- [x] Clean up new code
- [x] Resolve conflicts
- [x] Document new features

### CODE REVIEW

File: pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py [hunk 1]
```diff
@@ -230,7 +214,7 @@ def CropLatLon(d, regions_specs):
     """
     region_name = list(regions_specs.keys())[0]
 
-    dnew = region_subset(d, regions_specs, region=region_name)
+    dnew = region_subset(d, region_name, None, regions_specs, False)
```
Comment (lee1043) [inline pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py:217]: You might want to double check this with the function definition at https://github.com/PCMDI/pcmdi_metrics/blob/30640f10064d4bd8b1a4ee649cef76ae5fe2ffb3/pcmdi_metrics/io/regions.py#L82

My suggestion for this line for more clarification:

`dnew = region_subset(d, region_name, regions_specs=regions_specs)`

**Comment by acordonez** [comment]
Ready to review but running into CI issues due to same issue documented here: https://github.com/PCMDI/pcmdi_metrics/pull/1103

**Comment by lee1043** [comment]
@acordonez could you remind me an example commend to test this capability? Do we have a shape file example as a part of the demo download file?

**Comment by acordonez** [comment]
> @acordonez could you remind me an example commend to test this capability? Do we have a shape file example as a part of the demo download file?

@lee1043 I am not aware of a shapefile in the demo file. I've been running this with an NCA5 regions shapefile that I can point you to on perlmutter for testing.

**Comment by lee1043** [comment]
@acordonez I am running Demo 7 notebook with this branch, and getting below error. Could this be related to the change in this PR?

```
demo_data_tmp/CMIP5_demo_timeseries/historical/atmos/day/pr/
pr_day_GISS-E2-H_historical_r6i1p1_*.nc
[2000, 2005]
730 365
2
demo_output_tmp/precip_variability/GISS-E2-H/
demo_output_tmp/precip_variability/GISS-E2-H/
demo_output_tmp/precip_variability/GISS-E2-H/
['demo_data_tmp/CMIP5_demo_timeseries/historical/atmos/day/pr/pr_day_GISS-E2-H_historical_r6i1p1_20000101-20051231.nc']
GISS-E2-H.r6i1p1
['demo_data_tmp/CMIP5_demo_timeseries/historical/atmos/day/pr/pr_day_GISS-E2-H_historical_r6i1p1_20000101-20051231.nc']
GISS-E2-H.r6i1p1 365_day
syr, eyr: 2000 2005
Complete regridding from (2190, 90, 144) to (2190, 90, 180)
/home/lee1043/.conda/envs/pcmdi_metrics_dev_20240220/lib/python3.10/site-packages/pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py:315: FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead.
  climtime = pd.period_range(
Complete calculating climatology and anomaly for calendar of 365_day
Complete power spectra (segment:  730  nps: 5.0 )
Traceback (most recent call last):
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20240220/bin/variability_across_timescales_PS_driver.py", line 64, in <module>
    precip_variability_across_timescale(
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20240220/lib/python3.10/site-packages/pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py", line 87, in precip_variability_across_timescale
    psdmfm_forced = Avg_PS_DomFrq(ps, freqs, ntd, dat, mip, "forced", regions_specs)
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20240220/lib/python3.10/site-packages/pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py", line 496, in Avg_PS_DomFrq
    if "Ocean" in dom or regions_specs[dom].get("value", -1) == 0:
TypeError: 'NoneType' object is not subscriptable
```

**Comment by acordonez** [comment]
@lee1043 I've pushed a fix that I think should resolve this issue (it worked on the command line on perlmutter for me).

**Comment by lee1043** [comment]
> @lee1043 I've pushed a fix that I think should resolve this issue (it worked on the command line on perlmutter for me).

Thanks, it helped resolving the aforementioned error. I am continuing to test the rest part of demo 7, and once they all run without error, I will try the shape file, and keep you updated.

**Comment by lee1043** [comment]
In demo 7, it looks like there was a naming change, did I get that right?:
`PS_pr.day_regrid.180x90_area.freq.mean_GISS-E2-H.r6i1p1.json` to `PS_pr.day_regrid.3x3_area.freq.mean_GISS-E2-H.r6i1p1.json` (180x90 to 3x3)

@acordonez Could you please double check running demo 7 all cells on your end and make any necessary updates?

**Comment by acordonez** [comment]
@lee1043 The file should not be renamed, that was a bug. I've pushed the fix and the notebook should run without error now.

**Comment by lee1043** [comment]
Confirmed that demo notebook 7 works fine with the PR.

**Review by lee1043 — APPROVED** [review]

**Pull request merged**